### PR TITLE
W-7443194: Phase 2 of multi-engine story.  Collection of fixes and better support for tsconfig.

### PR DESCRIPTION
**Pull request by michaelhoefer**

### CODE REVIEW

**Comment by michaelhoefer** [comment]
Improved eslint/multi-engine support.  Far fewer changes than last time.  Still need to add some unit tests.

File: test/lib/RuleManager.test.ts [hunk 1]
```diff
@@ -148,4 +150,119 @@ describe('RuleManager', () => {
 			});
 		});
 	});
+
+	describe('runRulesMatchingCriteria()', () => {
```
Comment (jfeingold35) [inline test/lib/RuleManager.test.ts:154]: None of these tests call `runRulesMatchingCriteria`, though I assume that's what you're still working on.

File: src/lib/DefaultRuleManager.ts [hunk 2]
```diff
@@ -68,11 +66,15 @@ export class DefaultRuleManager implements RuleManager {
...
-		this.logger.trace(`Recombining results into requested format ${format}`);
-		return RuleResultRecombinator.recombineAndReformatResults(results, format);
+		try {
+			const psResults: RuleResult[][] = await Promise.all(ps);
+			psResults.forEach(r => results = results.concat(r));
+			this.logger.trace(`Received rule violations: ${results}`);
+			this.logger.trace(`Recombining results into requested format ${format}`);
+			return RuleResultRecombinator.recombineAndReformatResults(results, format);
+		} catch (e) {
+			throw new SfdxError(e.stack || e);
```
Comment (jfeingold35) [inline src/lib/DefaultRuleManager.ts:76]: Will we need to wrap the stack in something helpful to explain "Problem occurred while trying to do X, here's the stack"?

File: test/lib/RuleManager.test.ts (hunk 1 above)
Comment (michaelhoefer) [reply test/lib/RuleManager.test.ts:154]: Oops #1. Yes, that is in progress.  Was just deciding where to put the tests, landed on mirroring getRulesMatchingCriteria

File: src/lib/DefaultRuleManager.ts (hunk 2 above)
Comment (michaelhoefer) [reply src/lib/DefaultRuleManager.ts:76]: Oops #2.  I didn't mean to check that in, just changed it off the cuff to see the actual error info.

File: test/lib/RuleManager.test.ts (hunk 1 above)
Comment (michaelhoefer) [reply test/lib/RuleManager.test.ts:154]: Alright, I was able to add a simple test to run.test.ts as well as a few more here in RuleManager.test.ts.  I realized that I did not have to mock process.cwd().  I just needed to use process.chdir() to actually change it.  Duh.

**Comment by michaelhoefer** [comment]
@jfeingold35 ready to re-review.  Also added a JSON output format option, btw.

File: src/commands/scanner/run.ts [hunk 3]
```diff
@@ -174,7 +176,10 @@ export default class Run extends ScannerCommand {
 			// Default properly, again, as we did earlier.
 			const format: OUTPUT_FORMAT = this.flags.format || OUTPUT_FORMAT.TABLE;
 			// If we're just supposed to dump the output to the console, what precisely we do depends on the format.
-			if (format === OUTPUT_FORMAT.CSV && typeof output === 'string') {
+			if (format === OUTPUT_FORMAT.JSON && typeof output === 'string') {
+				// JSON is also just one giant string that we can dump directly to the console.
```
Comment (jfeingold35) [inline src/commands/scanner/run.ts:180]: This is probably the pettiest nit I've ever picked, but the 'also' should move to line 183.

File: src/lib/DefaultRuleManager.ts (hunk 2 above)
Comment (jfeingold35) [reply src/lib/DefaultRuleManager.ts:76]: Should this still be here?

File: src/commands/scanner/run.ts (hunk 3 above)
Comment (michaelhoefer) [reply src/commands/scanner/run.ts:180]: It shows you care, @jfeingold35

File: src/lib/DefaultRuleManager.ts (hunk 2 above)
Comment (michaelhoefer) [reply src/lib/DefaultRuleManager.ts:76]: Gone.

**Pull request merged**